src/process/path.h

```diff
@@ -154,6 +199,25 @@ static __always_inline int write_path(struct pt_regs *ctx, cached_path_t *cached
         if (bpf_probe_read(&offset, sizeof(offset), dentry + name) < 0)
             goto NameError;
 
+        // Path filtering, only if the filter is active
+        if (cached_path->filter_state >= 0) {
+            val = NULL;
+            // We need to reset at least the whole string storage array each time we use the key
+            bpf_probe_read(&key, sizeof(filter_key_t), &blank_key);
```

Comment:
is this easier/less instructions than a `key = (filter_key_t){0}`?

Reply:
Since helper functions aren't inlined, I suspect it's far fewer instructions to call a helper than write 0s into the whole struct.  Would have to compare generated code to be sure, though.